### DISCUSSION TITLE
IRAF installation -xgterm-

### DISCUSSION THREAD

**Discussion by UrosM13** [iraf-community/iraf#212] [Q&A] [answered]
Hi,
I am using IRAF few years but this is the first time I am trying to install it.
Installation is done on iMac (24-inch, M1, 2021) macOS Monterey version 12.1

So, installed IRAF following instructions from here (https://github.com/iraf-community/iraf/blob/main/INSTALL.md).
Everything went ok, no problems.

I tried to run in terminal $ xgterm &  (this is what usually I am doing on my old computer and then cl)
But I get this message  zsh: command not found: xgterm

Please can someone give me directions on how to solve this? 
(I think the problem is related to .bash_profile, so I am attaching how my bash profile looks like at the moment. )

Thanks :)

Uros

<img width="894" alt="Screenshot 2022-02-18 at 19 29 44" src="https://user-images.githubusercontent.com/93936495/154741974-c6be1f1b-e09d-481f-a871-f113edfd3e43.png">

**Comment by olebole** [comment, marked as answer]
You also need to install xgterm from the x11iraf package. For this, follow the [installation instructions](https://iraf-community.github.io/x11iraf.html).

**Comment by UrosM13** [comment]
Hi, 

Finally, I find time to continue with IRAF installation :)

I think xgterm is now installed properly. After typing in terminal & xgterm & I am getting a new window (as expected).
Now I am typing & cl to start IRAF but this is what I get   zsh: command not found: cl
(also image attached)

Any suggestions on what could be wrong and how to fix it, Please :)


Thanks


Uros



<img width="502" alt="Screenshot 2022-02-25 at 17 42 35" src="https://user-images.githubusercontent.com/93936495/155753540-18731407-e214-4222-9e52-f742076af894.png">

**Reply by olebole** [reply]
The "cl" script must be in your `PATH`, so please check whether it contains `/Users/uros/.iraf/bin`, and (if yes) that there is a `cl` symlink pointing to `$iraf/unix/hlib/cl.sh`. You can also try to call the latter directly. Note however, that this is plain cl, i.e. without command line editing. You probably want the "ecl", and should then follow the same way.

**Reply by UrosM13** [reply]
Hi, 

Thanks for the reply.

I check PATH and I think it contains  /Users/uros/.iraf/bin. (image attached).

The second part regarding symlink is not clear to me ... do I need to create symlink or how to check if there is a symlink pointing to $iraf/unix/hlib/cl.sh .


My apology for bothering you, probably with some basic stuff. But for me all this is new.


<img width="883" alt="Screenshot 2022-02-28 at 11 18 44" src="https://user-images.githubusercontent.com/93936495/155966933-b94b89d9-71e8-4cd4-a840-28a43726f33b.png">


Thanks.

Uros

**Reply by olebole** [reply]
This path should usually be set in `~/.iraf/setup.sh`, so you don't need the last line in your file. However, ma question is whether it is really set in your shell. Do you see it when you do a `echo $PATH`?. 
And, you should check the contents of `~/.iraf/bin/`, especially if there is a link `cl --> …/unix/hlib/cl.sh`. What is the output of `ls -l ~/.iraf/bin`?

**Reply by UrosM13** [reply]
After doing echo $PATH, yes I can see it (also img attached).
<img width="990" alt="Screenshot 2022-02-28 at 12 49 27" src="https://user-images.githubusercontent.com/93936495/155978677-9375a81c-4c9c-450d-b777-178e9eca9040.png">

Output of the ls -l ~/.iraf/bin   img. attached.
<img width="989" alt="Screenshot 2022-02-28 at 12 51 43" src="https://user-images.githubusercontent.com/93936495/155978931-dbcc8092-2bd4-45c1-aaac-930a10304b4b.png">


Attached is also cl file from location /.iraf/bin/   (I just added .txt extension so it can be attached)
[cl.txt](https://github.com/iraf-community/iraf/files/8153104/cl.txt)


Thank you for helping me :)

**Reply by olebole** [reply]
Did you run these commands (especially `echo $PATH` within the zsh or within the bash (i.e. did you do this within the xgterm)? What happens if you run `~/Desktop/iraf-main/unix /hlib/cl.sh` directly?

**Reply by UrosM13** [reply]
I run those commands within zsh.

This is what I am getting when running both commands within xgterm (img attached).
<img width="811" alt="Screenshot 2022-02-28 at 13 10 31" src="https://user-images.githubusercontent.com/93936495/155981408-c569cb21-6092-4b78-b11a-60626e46d37d.png">

After running ~/Desktop/iraf-main/unix/hlib/cl.sh. (within zsh) I am getting this (img attached).
<img width="1000" alt="Screenshot 2022-02-28 at 13 14 54" src="https://user-images.githubusercontent.com/93936495/155981867-385671f3-e3d0-40f2-aa57-6f10cba78229.png">

**Reply by olebole** [reply]
That shows that you should just add the directory to your PATH in the zsh startup file.

**Reply by UrosM13** [reply]
Excellent, all works now :)

Is it posssible to install some external packages to IRAF, like RVSAO.

Thanks for help.

All the best.


Uros

**Reply by olebole** [reply]
Yes it is. This is done f.e. in Debian and Ubuntu: https://packages.debian.org/bullseye/iraf-rvsao and this also is the easiest way.
For other systems, it is not that easy; volunteers to make this simpler are welcome.
(I always wonder why astronomers insist in using MacOS, where the installation of basic astronomical packages is a pain, compared with Debian - SCNR)

**Reply by UrosM13** [reply]
Thanks :)
At the moment I am working on MacOS so need to find a way to install it on Mac ....
If happens that I improve my knowledge on that level to be able to make installation simple on other systems will be happy to contribute in the future.

(Not sure why MacOS is chosen by the astronomical community ... )

**Reply by olebole** [reply]
You could also ask the author of RVSAO, @jessmink (Jessica Mink, [SAO](http://tdc-www.harvard.edu/)).

**Reply by UrosM13** [reply]
Hi,

I managed to install RVSAO (thnks for suggesting to contact Jessica Mink).
Do anyone knows how to fix > ERROR: Cannot open connected subprocess (./x_rvsao.e) <
It happens when I am trying to run xcsao task.

Thanks.

Uros

**Reply by olebole** [reply]
Maybe you ask Jessika about this? She is the best person who knows the details of RVSAO.

**Reply by UrosM13** [reply]
Will try :)

Thanks.
